src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepExecution.java

```diff
@@ -53,39 +58,83 @@
     @Override
     public boolean start() throws Exception {
         String job = step.getJob();
-        final ParameterizedJobMixIn.ParameterizedJob project = Jenkins.getActiveInstance().getItem(job, invokingRun.getParent(), ParameterizedJobMixIn.ParameterizedJob.class);
-        if (project == null) {
-            throw new AbortException("No parameterized job named " + job + " found");
+        Item item = Jenkins.getActiveInstance().getItem(job, invokingRun.getParent(), Item.class);
+        if (item == null) {
+            throw new AbortException("No item named " + job + " found");
         }
-        listener.getLogger().println("Scheduling project: " + ModelHyperlinkNote.encodeTo(project));
-
-        node.addAction(new LabelAction(Messages.BuildTriggerStepExecution_building_(project.getFullDisplayName())));
-        List<Action> actions = new ArrayList<>();
-        if (step.getWait()) {
-            StepContext context = getContext();
-            actions.add(new BuildTriggerAction(context, step.isPropagate()));
-            LOGGER.log(Level.FINER, "scheduling a build of {0} from {1}", new Object[] {project, context});
-        }
-        actions.add(new CauseAction(new Cause.UpstreamCause(invokingRun)));
-        List<ParameterValue> parameters = step.getParameters();
-        if (parameters != null) {
-            parameters = completeDefaultParameters(parameters, (Job) project);
-            actions.add(new ParametersAction(parameters));
+        if (step.getWait() && !(item instanceof Job)) {
+            // TODO find some way of allowing ComputedFolders to hook into the listener code
+            throw new AbortException("Waiting for non-job items is not supported");
```

Comment:
Note that this kind of condition could also be checked in `doCheck*` methods for the benefit of **Pipeline Syntax**.